.tito/lib/git_annex_spec_builder.py

```diff
@@ -0,0 +1,168 @@
+        avoid using these if possible.  *Given in the format of a dictionary
+        of lists.*
+        """
+        ConfigObject.__init__(self, config=config)
+        BuilderBase.__init__(self, name=name, build_dir=build_dir, config=config,
+                user_config=user_config, args=args, **kwargs)
+        self.build_tag = tag
+
+        self.build_version = self._get_build_version()
+        self.git_commit_id = self.build_version
+
+        if kwargs and 'options' in kwargs:
+            warn_out("'options' no longer a supported builder constructor argument.")
+
+        if self.config.has_section("requirements"):
+            if self.config.has_option("requirements", "tito"):
+                if loose_version(self.config.get("requirements", "tito")) > \
+                        loose_version(require('tito')[0].version):
+                    error_out([
+                        "tito version %s or later is needed to build this project." %
+                        self.config.get("requirements", "tito"),
+                        "Your version: %s" % require('tito')[0].version
+                    ])
+
+        self.display_version = self._get_display_version()
+
+        self.relative_project_dir = get_relative_project_dir_cwd(self.git_root)
+
+        tgz_base = self._get_tgz_name_and_ver()
+        self.tgz_filename = tgz_base + ".tar.gz"
+        self.tgz_dir = tgz_base
+        self.artifacts = []
+
+        self.rpmbuild_gitcopy = os.path.join(self.rpmbuild_sourcedir,
+                self.tgz_dir)
+
+        # Used to make sure we only modify the spec file for a test build
+        # once. The srpm method may be called multiple times during koji
+        # releases to create the proper disttags, but we only want to modify
+        # the spec file once.
+        self.ran_setup_test_specfile = False
+
+        # NOTE: These are defined later when/if we actually dump a copy of the
+        # project source at the tag we're building. Only then can we search for
+        # a spec file.
+        self.spec_file_name = None
+        self.spec_file = None
+
+        # Set to path to srpm once we build one.
+        self.srpm_location = None
+
+    def _get_build_version(self):
+        self.spec_version = ''
+        self.spec_release = ''
+        build_version = get_spec_version_and_release(
+            self.start_dir,
+            find_spec_like_file(self.start_dir)
+        )
+        self.build_tag = build_version
+        return build_version
+
+    def _setup_sources(self):
+        """
+        Create a copy of the git source for the project at the point in time
+        our build tag was created.
+
+        Created in the temporary rpmbuild SOURCES directory.
+        """
+        self._create_build_dirs()
+
+        debug('SETUP SOURCES')
+        working_path = os.getcwd() if self.relative_project_dir in os.path.join(os.getcwd(), '') else os.path.join(os.getcwd(), self.relative_project_dir)
```

Comment:
This is odd code. A path check with string comparison? Shouldn't this be ```if self.relative_project_dir == os.path.basename(os.getcwd())```?

Reply:
Maybe? I can certainly change it, but this was taken from `tito` code serving as my example.

Reply:
So, turns out that `relative_project_dir` is a partial path and the check is for whether that partial path is contained within the full path specified by `os.path.join(os.getcwd(), '')` because of how tito sets up the source environment.